js/saveState.js

```diff
@@ -0,0 +1,42 @@
+import _ from "lodash";
+
+import { COOKIE_LAST_STATE, COOKIE_OMIT_FROM_STATE } from "./constants/AppConstants";
+
+// store the cookie here in case we're testing
+let lastCookie = "";
+
+function cookiesToMap (cookieStr) {
+  return _.chain(cookieStr.split(/;\s*/))
+          .map((keyValStr) => keyValStr.split(/=/))
+          .fromPairs()
+          .value();
+}
+
+export function restoreState() {
+	let cookie = lastCookie;
+	if (typeof document !== 'undefined') {
+    cookie = document.cookie;
+	}
+  const stateEncoded = cookiesToMap(cookie)[COOKIE_LAST_STATE];
+  if (stateEncoded) {
+    const stateJson = decodeURIComponent(stateEncoded);
+    return JSON.parse(stateJson);
+  } else {
+    return {};
+  }
+}
+
+export function saveState(state) {
+	const stateToSave = _.omit(state, COOKIE_OMIT_FROM_STATE);
+	const stateJson = JSON.stringify(stateToSave);
+	const stateEncoded = encodeURIComponent(stateJson);
+	const stateCookie = COOKIE_LAST_STATE + "=" + stateEncoded;
```

Comment:
Hmm, it looks like the `atob` and `btoa` functions referenced in that MDN article aren't available in the node testing environment. I'm sure there's a library that could provide them as a polyfill but it doesn't seem worth the trouble if URL encoding works fine. Unless you feel strongly about this I'm going leave it as URL encoding.